app/assets/javascripts/discourse/models/category.js.es6

```diff
@@ -47,9 +47,9 @@ const Category = RestModel.extend({
     return Discourse.getURL("/c/") + Category.slugFor(this);
   },
 
-  @computed("url")
-  fullSlug(url) {
-    return url.slice(3).replace("/", "-");
```

Comment:
This does not work well with subdirectory installations (i.e. `/forum/c/my-category` is transformed to `rum-c/my-category`).

Using `replace(/\//g, "-")` would partially fix (transform to `rum-c-my-category`), but it is still not very good.

Reply:
We have `Discourse.BaseUri` available so you should be able to do

```javascript
@computed
fullSlug() {
  return Category.slugFor(this).slice(Discourse.BaseUri.length + 3).replace(\\//g, "-");
}
```

Reply:
That looked hacky imo, but I agree we should do it like that to be the same with the logic on the server-side.

Reply:
Why is this "hacky"? This is the same logic as [server-side](https://github.com/discourse/discourse/pull/6622/files#diff-af69b723432608c351fb1dd5db9e8348R538).

Reply:
> Well... `url == Discourse.getURL("/c/") + Category.slugFor(this)`

If that's **always** the case, then fine 👍 Might need a comment.



> There is also that `3` which is a magic constant (stands for `'/c/'.length`).

I agree, kill all the magic constants ;)